BedrockServer.cpp

```diff
@@ -828,6 +812,34 @@ void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlo
 
     int64_t lastConflictPage = 0;
     while (true) {
+
+        // If there are outstanding HTTPS requests on this command (from a previous call to `peek`) we process them here.
+        while (!command->areHttpsRequestsComplete()) {
+            SINFO("Running command network activity loop.");
+            fd_map fdm;
+            command->prePoll(fdm);
+            const uint64_t now = STimeNow();
+            uint64_t nextActivity = 0;
+            S_poll(fdm, max(nextActivity, now) - now);
+
+            // Timeout is five minutes unless we're shutting down or standing down, in which case it's 5 seconds.
+            // Note that BedrockCommad::postPoll sets the timeout to the command's timeout if it's lower than this value anyway,
+            // So this only has an effect if it will be shorter than the command's timeout.
+            uint64_t maxWaitMs = 5 * 60 * 1'000;
+            auto _syncNodeCopy = atomic_load(&_syncNode);
+            if (_shutdownState.load() != RUNNING || (_syncNodeCopy && _syncNodeCopy->getState() == SQLiteNodeState::STANDINGDOWN)) {
+                maxWaitMs = 5'000;
+            }
+
+            auto start = STimeNow();
+            command->postPoll(fdm, nextActivity, maxWaitMs);
+            auto elapsedUS = STimeNow() - start;
+            if (elapsedUS > 100'000) {
+                // We warn here as this is a potential serious performance issue that seems to happen sometimes.
+                SWARN("Post poll on command '" << command->request.methodLine << "' took " << elapsedUS << "us.");
+            }
+        }
```

Comment:
This whole block is a simpler version of `_postPollCommands` which is now gone. It's actually much cleaner here and doesn't require us to do nearly as much bookkeeping to know what commands are where.